modules/steam/autostart.nix

```diff
@@ -31,7 +31,15 @@ in
         };
 
         desktopSession = mkOption {
-          type = types.nullOr types.str;
+          type = with types ; nullOr str // {         
+            check = userProvidedDesktopSession:
+              lib.assertMsg  (str.check userProvidedDesktopSession && userProvidedDesktopSession ? config.services.xserver.displayManager.sessionData.sessionNames) ''
+                  Desktop session, '${userProvidedDesktopSession}', not found.
```

Comment:
Nit: the commas don't really work without "Default" at the start, so maybe drop them.